src/contracts/BCoWHelper.sol

```diff
@@ -117,9 +97,88 @@ contract BCoWHelper is ICOWAMMPoolHelper, BMath {
     if (tokens_.length != 2) {
       revert PoolDoesNotExist();
     }
-    // reverts in case pool is not supported (non-equal weights)
-    if (IBCoWPool(pool).getNormalizedWeight(tokens_[0]) != IBCoWPool(pool).getNormalizedWeight(tokens_[1])) {
-      revert PoolDoesNotExist();
+  }
+
+  /// @notice Returns the order that is suggested to be executed to CoW Protocol
+  /// for a specific pool given the current chain state and final token prices
+  /// @param pool The pool whose funds should be used in the order
+  /// @param prices The prices for each token of the pool
+  /// @return order_ The suggested CoW Protocol order
+  function _rawOrder(address pool, uint256[] calldata prices) internal view returns (GPv2Order.Data memory order_) {
+    address[] memory tokenPair = tokens(pool);
+    Reserves memory reservesOut = _reserves(IBCoWPool(pool), IERC20(tokenPair[0]));
+    Reserves memory reservesIn = _reserves(IBCoWPool(pool), IERC20(tokenPair[1]));
+
+    // The price of this function is expressed as amount of token1 per amount
+    // of token0. The `prices` vector is expressed the other way around, as
+    // confirmed by dimensional analysis of the expression above.
+    uint256 priceNumerator = prices[1]; // x token = sell token = out amount
+    uint256 priceDenominator = prices[0];
+    uint256 balanceOutTimesWeightIn = bmul(reservesOut.balance, reservesIn.normWeight);
+    uint256 balanceInTimesWeightOut = bmul(reservesIn.balance, reservesOut.normWeight);
+    // This check compares the (weight-adjusted) pool spot price with the input
+    // price.
+    if (bmul(balanceInTimesWeightOut, priceNumerator) > bmul(balanceOutTimesWeightIn, priceDenominator)) {
+      (reservesOut, reservesIn) = (reservesIn, reservesOut);
+      (balanceOutTimesWeightIn, balanceInTimesWeightOut) = (balanceInTimesWeightOut, balanceOutTimesWeightIn);
+      (priceNumerator, priceDenominator) = (priceDenominator, priceNumerator);
     }
+    // The out amount is computed according to the following formula:
+    // aO = amountOut
+    // bI = reservesIn.balance                   bO * wI - p * bO * wI
+    // bO = reservesOut.balance            aO =  ---------------------
+    // wI = reservesIn.denormWeight                     wI + wO
```

Comment:
Can you add some comments to clarify that because `wI + wO = 1`, it can be excluded from the calculation below.